test/lib/tokenFactory.test.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const assert = require('chai').assert;
+const mockery = require('mockery');
+const sinon = require('sinon');
+
+sinon.assert.expose(assert, { prefix: '' });
+require('sinon-as-promised');
+
+describe('Token Factory', () => {
+    const name = 'mobile_token';
+    const description = 'a token for a mobile app';
+    const uuid = 'abc123';
+    const userId = 6789;
+    const tokenId = 12345;
+    const tokenData = {
+        id: tokenId,
+        userId,
+        description,
+        name,
+        uuid,
+        lastUsed: null
+    };
+    let TokenFactory;
+    let datastore;
+    let factory;
+    let Token;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        datastore = {
+            save: sinon.stub()
+        };
+
+        /* eslint-disable global-require */
+        Token = require('../../lib/token');
```

Comment:
I see it now. You're getting that because you'll also need to mock the `Token` class as well.

This is fine as-is.